lib/date_time/date_time_core.c

```diff
@@ -218,6 +218,11 @@ void date_time_core_init(void)
 	if (!IS_ENABLED(CONFIG_DATE_TIME_AUTO_UPDATE)) {
 		date_time_core_schedule_update();
 	}
+
+	if (IS_ENABLED(CONFIG_BOARD_NATIVE_SIM)) {
+		LOG_DBG("Native SIM detected, consider time updated.");
```

Comment:
Uppercase "SIM" is easily confused with a SIM card, I would rather use "sim" or "simulator" instead.